WordPress/Classes/Models/AbstractPost.swift

```diff
@@ -146,4 +156,58 @@ extension AbstractPost {
     func hasPermanentFailedMedia() -> Bool {
         return media.first(where: { !$0.willAttemptToUploadLater() }) != nil
     }
+
+    /// Returns a list of changes from the current post to the previous revision.
+    var changes: RemotePostUpdateParameters {
+        guard let original else {
+            return RemotePostUpdateParameters() // Empty
+        }
+        return RemotePostUpdateParameters.changes(from: original, to: self)
+    }
+
+    var isEmpty: Bool {
+        let title = (postTitle ?? "").trimmingCharacters(in: .whitespaces)
+        let content = (content ?? "").trimmingCharacters(in: .whitespaces)
+        return title.isEmpty && content.isEmpty
+    }
+
+    // TODO: Replace with a new flag
+    @objc var isSyncNeeded: Bool {
```

Comment:
I'm planning to add a new flag to the database but only after we merge the changes in `trunk` and after 24.6 is cut. We'll be also removing some data.